src/dataset.rs

```diff
@@ -174,6 +174,7 @@ impl GeoTransformEx for GeoTransform {
 #[derive(Debug)]
 pub struct Dataset {
     c_dataset: GDALDatasetH,
+    closed: bool,
```

Comment:
We could also have a wrapper around `c_dataset` and use `ManuallyDrop` on that instead.

Reply:
`ManuallyDrop` sounds nice

Reply:
I switched to `ManuallyDrop` in https://github.com/georust/gdal/commit/2e4b75ee6c974febd7da09cd8859aecbb77674c5, but I'm not sure it's much better.

Reply:
Scratch that, it doesn't really work, because the inner type's `drop` still runs, and we can't get the return value of `GDALClose` out of there. I'll revert to the previous version.